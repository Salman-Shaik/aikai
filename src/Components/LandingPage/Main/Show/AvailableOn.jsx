import Tooltip from "@material-ui/core/Tooltip";
import React from "react";

export const AvailableOn = ({ homePage }) => {
  const isNetflix = () => homePage.includes("netflix");
  const isDisneyPlus = () => homePage.includes("disney");
  const isPrimeVideo = () =>
    homePage.includes("amazon") || homePage.includes("primevideo");
  const goToHomepage = () => window.open(homePage);

  return isNetflix() || isPrimeVideo() || isDisneyPlus() ? (
    <section className="available_on">
      <h4>Available On: </h4>
      {isNetflix() && (
        <Tooltip title="May Require Subscription.">
          <img
            src={`/Netflix.png`}
            alt="Netflix"
            className="stream"
            onClick={goToHomepage}
          />
        </Tooltip>
      )}
      {isPrimeVideo() && (
        <Tooltip title="May Require Subscription.">
          <img
            src={`/Prime.png`}
            alt="Prime video"
            className="stream prime"
            onClick={goToHomepage}
          />
        </Tooltip>
      )}
      {isDisneyPlus() && (
        <Tooltip title="May Require Subscription.">
          <img
            src={`/Disney+.jpg`}
            alt="Disney +"
            className="stream prime"
            onClick={goToHomepage}
          />
        </Tooltip>
      )}
    </section>
  ) : (
    <span />
  );
};
