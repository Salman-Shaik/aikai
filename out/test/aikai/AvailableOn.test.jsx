import { render } from "@testing-library/react";
import React from "react";
import { AvailableOn } from "../AvailableOn";

describe("Available On", () => {
  it("Snapshot Test", function () {
    const { container } = render(<AvailableOn homePage="" />);
    expect(container).toMatchSnapshot();
  });

  it("should return a empty span when the homepage is not recognized or empty", () => {
    const { container } = render(<AvailableOn homePage="" />);
    const emptySpan = container.querySelector(`span`);
    expect(emptySpan).toBeDefined();
  });

  it("should return Netflix Image when the homepage is netflix link", () => {
    const { getByAltText } = render(<AvailableOn homePage="netflix.com" />);
    const netflixImage = getByAltText("Netflix");
    expect(netflixImage).toBeDefined();
  });

  it("should return Prime Image when the homepage is netflix link", () => {
    const { getByAltText } = render(<AvailableOn homePage="primevideo.com" />);
    const primeImage = getByAltText("Prime video");
    expect(primeImage).toBeDefined();
  });

  it("should return Disney Image when the homepage is netflix link", () => {
    const { getByAltText } = render(<AvailableOn homePage="disney.com" />);
    const disneyImage = getByAltText("Disney +");
    expect(disneyImage).toBeDefined();
  });
});
