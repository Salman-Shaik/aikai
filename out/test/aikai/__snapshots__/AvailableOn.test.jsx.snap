// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Available On Snapshot Test 1`] = `
<div>
  <span />
</div>
`;
